blocks/api/parser.js

```diff
@@ -88,6 +53,69 @@ export function asType( value, type ) {
 }
 
 /**
+ * Returns an hpq matcher given a source object
+ *
+ * @param  {Object}   sourceConfig Attribute Source object
+ * @return {Function}              hpq Matcher
+ */
+export function matcherFromSource( sourceConfig ) {
+	switch ( sourceConfig.source ) {
+		case 'attribute':
+			return attr( sourceConfig.selector, sourceConfig.attribute );
+		case 'property':
+			return prop( sourceConfig.selector, sourceConfig.property );
+		case 'html':
+			return html( sourceConfig.selector );
+		case 'text':
+			return text( sourceConfig.selector );
+		case 'children':
+			return children( sourceConfig.selector );
+		case 'node':
+			return node( sourceConfig.selector );
+		case 'query':
+			return query( sourceConfig.selector, matcherFromSource( sourceConfig.query ) );
+		case 'object':
+			return keys( sourceConfig.object ).reduce( ( memo, key ) => {
+				memo[ key ] = matcherFromSource( sourceConfig.object[ key ] );
+				return memo;
+			}, {} );
+		default:
+			// eslint-disable-next-line no-console
+			console.error( `Unkown source type "${ sourceConfig.source }"` );
+	}
+}
+
+/**
+ * Given a blocktype, a block's raw content and the commentAttributes returns the attribute value depending on its source definition
+ *
+ * @param  {string} attributeKey        Attribute key
+ * @param  {Object} attributeSchema     Attribute's schema
+ * @param  {string} rawContent          Block's raw content
+ * @param  {Object} commentAttributes   Block's comment attributes
+ *
+ * @return {mixed}                      Attribute value
```

Comment:
From what I can tell, there is no `mixed` type for JSDoc, but rather the `*` (any) type:

http://usejsdoc.org/tags-param.html#multiple-types-and-repeatable-parameters